python/ray/tune/tune.py

```diff
@@ -364,6 +365,12 @@ def run(
             "The `mode` parameter passed to `tune.run()` has to be one of "
             "['min', 'max']")
 
+    if scheduler is not None and isinstance(scheduler,
+                                            PopulationBasedTraining) and \
+            search_alg is not None:
+        raise ValueError("Search algorithms cannot be used with PBT or PB2 "
+                         "schedulers. Please remove {}.".format(search_alg))
+
```

Comment:
Can we instead raise this in pbt.py, like in `on_trial_add`?

Reply:
My concern with that is that won't that actually attempt to start training, and then raise errors in each remote training processes? That could possibly result in confusing errors.

This is a usage error with `tune.run` so I thought we can preemptively raise an error on the driver before training even begins.

Reply:
No, on_trial_add is called before any remote processes start, so it should be safe to raise an error there

Reply:
Ah ok, got it. I pushed a change for it.